app/containers/Demo/reducer.js

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ * Demo reducer
+ *
+ */
+import produce from 'immer';
+import { createActions } from 'reduxsauce';
+import { get } from 'lodash';
+
+export const initialState = {
+  songs: [],
+  error: '',
+  loading: false,
+  searchText: ''
+};
+
+export const { Types: demoTypes, Creators: demoCreators } = createActions({
+  getSongs: ['searchText'],
+  successGetSongs: ['data'],
+  errorGetSongs: ['error']
+});
+
+/* eslint-disable default-case, no-param-reassign */
+export const demoReducer = (state = initialState, action) =>
+  produce(state, draft => {
+    switch (action.type) {
+      case demoTypes.GET_SONGS:
```

Comment:
typically we don't want to put any business logic in the reducer, we want it to be as simple as possible. 

The intent here makes sense, let's move this business logic to the saga